brian2/units/fundamentalunits.py

```diff
@@ -460,7 +458,11 @@ def __ipow__(self, value):
 
     #### COMPARISON ####
     def __eq__(self, value):
-        return np.allclose(self._dims, value._dims)
+        try:
```

Comment:
If we change the is checks to use an ``==`` check instead, we could first check if ``self._dims is value._dims`` and potentially get most of the speed still.